### PR TITLE
Add a --remove option to fix-encoding-pragma

### LINKED ISSUE #106: Add a --remove option to fix-encoding-pragma
This is an unnecessary thing in python3

### PR DESCRIPTION
**Pull request by asottile**
Resolves #106

### CODE REVIEW

File: pre_commit_hooks/fix_encoding_pragma.py [hunk 1]
```diff
@@ -21,53 +21,98 @@ def has_coding(line):
...
-    old = f.read()
-    f.seek(0)
+class ExpectedContents(collections.namedtuple(
+        'ExpectedContents', ('shebang', 'rest', 'pragma_status'),
+)):
+    """
+    pragma_status:
+    - True: has exactly the coding pragma expected
+    - False: missing coding pragma entirely
+    - None: has a coding pragma, but it does not match
```
Comment (chriskuehl) [inline pre_commit_hooks/fix_encoding_pragma.py]:
```c++
enum Bool 
{ 
    True, 
    False, 
    FileNotFound 
};
```

File: pre_commit_hooks/fix_encoding_pragma.py [hunk 2]
```diff
@@ -21,53 +21,98 @@ def has_coding(line):
...
+        else:
+            return 0
+
+    if expected.is_expected_pragma(remove):
+        return 0
+
+    # Otherwise, write out the new file
+    f.seek(0)
+    f.truncate()
+    f.write(expected.shebang)
```
Comment (chriskuehl) [inline pre_commit_hooks/fix_encoding_pragma.py]: i feel like you told me this doesn't work on Windows

**Comment by chriskuehl** [comment]
shipit!

File: pre_commit_hooks/fix_encoding_pragma.py (hunk 2 above)
Comment (asottile) [inline pre_commit_hooks/fix_encoding_pragma.py]: ah that was using `NamedTemporaryFile` and then passing the file handle to anther process which writes to it :D

**Pull request merged**